src/components/Upload.tsx

```diff
@@ -29,53 +75,13 @@ export default function Upload({ setInUploadView }: UploadProps) {
             rejectedFiles.map((fileWrapper) => `\n${fileWrapper.file.name}`)
         );
     },
-    []
+    [parse]
   );
   const { getRootProps, getInputProps } = useDropzone({
     onDrop,
     accept: ".dcm",
   });
 
-  const parse = (file: File): boolean => {
-    file.arrayBuffer().then((arrayBuffer) => {
```

Comment:
Przeniosłem to `useCallback`.
Mam pomysł na to jak można trochę ładniej napisać tę funkcję więc spróbuję w ten weekend i mam nadzieję, że się uda :D